src/components/apps/PublishAppDialog.js

```diff
@@ -0,0 +1,365 @@
+import ids from "./ids";
+import constants from "../../constants";
+import appsConstants from "./constants";
+import { useConfig } from "contexts/config";
+
+import {
+    announce,
+    AnnouncerConstants,
+    build,
+    FormMultilineTextField,
+    FormTextField,
+} from "@cyverse-de/ui-lib";
+
+import DEDialog from "components/utils/DEDialog";
+import ResourceTypes from "components/models/ResourceTypes";
+import FolderInput from "components/apps/launch/params/FolderInput";
+import { validateAppName, formatAppDoc } from "components/apps/utils";
+
+import { requestToPublishApp } from "serviceFacades/apps";
+
+import ErrorTypographyWithDialog from "components/utils/error/ErrorTypographyWithDialog";
+
+import { Add, Remove } from "@material-ui/icons";
+import {
+    Button,
+    CircularProgress,
+    Grid,
+    IconButton,
+    Typography,
+} from "@material-ui/core";
+
+export default function PublishAppDialog(props) {
+    const { open, app, handleClose } = props;
+    const { t } = useTranslation("apps");
+    const { t: i18nCommon } = useTranslation("common");
+    const { t: i18nUtil } = useTranslation("util");
+    const [config] = useConfig();
+    const communityPath = config?.irods?.community_path;
+    const parentId = ids.PUBLISH_DLG;
+    const [error, setError] = React.useState(null);
+
+    const [requestAppPublication, { status: requestStatus }] = useMutation(
+        requestToPublishApp,
+        {
+            onSuccess: () => {
+                announce({
+                    text: t("publicationRequestSuccess", {
+                        appName: app?.name,
+                    }),
+                    variant: AnnouncerConstants.INFO,
+                });
+                handleClose();
+            },
+            onError: setError,
+        }
+    );
+
+    const handleSubmit = (values) => {
+        const documentation = formatAppDoc(
+            values?.name,
+            values.description,
+            values.testData,
+            values?.inputDesc,
+            values?.parameterDesc,
+            values?.outputDesc
+        );
+        const request = {
+            description: values?.description,
+            documentation,
+            id: app?.id,
+            name: values?.name,
+            system_id: app?.system_id,
+            references: values?.references,
+        };
+        if (requestStatus !== constants.LOADING) {
+            requestAppPublication({
+                systemId: app?.system_id,
+                appId: app?.id,
+                request,
+            });
+        }
+    };
+
+    const validateTestDataPath = (value) => {
+        if (!value.includes(communityPath)) {
+            return t("testDataPathError");
+        }
+        return;
+    };
+
+    return (
+        <Formik
+            initialValues={{
+                name: app?.name,
+                description: app?.description,
+                system_id: app?.system_id,
```

Comment:
It doesn't look like the form is using `system_id` (the `handleSubmit` function is directly using `app?.system_id`).
```suggestion
```